great_expectations/checkpoint/checkpoint.py

```diff
@@ -99,6 +106,85 @@ def get_substituted_config(
                     self._substituted_config = substituted_config
                 return substituted_config
 
+    def get_runtime_batch_request(
+        self,
+        substituted_runtime_config: CheckpointConfig,
+        validation_batch_request: Union[dict, BatchRequest, None],
+    ) -> BatchRequest:
+        if substituted_runtime_config.batch_request is None:
+            return (
+                validation_batch_request
+                if isinstance(validation_batch_request, (BatchRequest, type(None)))
+                else BatchRequest(**validation_batch_request)
+            )
+
+        # get batch requests as dicts; since get_json_dict() doesn't include batch_data, re-add
+        base_batch_request_batch_data: Any = (
+            substituted_runtime_config.batch_request.batch_data
+        )
+        base_batch_request_dict: dict = (
+            substituted_runtime_config.batch_request.get_json_dict()
+        )
+        base_batch_request_dict["batch_data"] = base_batch_request_batch_data
+
+        if isinstance(validation_batch_request, BatchRequest):
+            validation_batch_request_batch_data: Any = (
+                validation_batch_request.batch_data
+            )
+            validation_batch_request: dict = validation_batch_request.get_json_dict()
+            validation_batch_request["batch_data"] = validation_batch_request_batch_data
+        elif isinstance(validation_batch_request, None):
+            return BatchRequest(**base_batch_request_dict)
+
+        runtime_batch_request_dict = deepcopy(validation_batch_request)
+        for key, val in runtime_batch_request_dict.items():
+            if val is not None and base_batch_request_dict.get(key) is not None:
+                raise CheckpointError(
+                    f"BatchRequest attribute '{key}' was specified in both validation and top-level CheckpointConfig."
+                )
+            runtime_batch_request_dict[key] = base_batch_request_dict[key]
+        return BatchRequest(**runtime_batch_request_dict)
+
+    def validate_validation_dict(self, validation_dict):
+        if validation_dict.get("batch_request") is None:
+            raise CheckpointError("validation batch_request cannot be None")
+        if not validation_dict.get("expectation_suite_name"):
+            raise CheckpointError("validation expectation_suite_name must be specified")
+        if not validation_dict.get("action_list"):
+            raise CheckpointError("validation action_list cannot be empty")
+
+    def get_substituted_validation_dict(
```

Comment:
Same here -- should this be a private method?

Reply:
Yep. Renaming.